crates/core/src/db/relational_db.rs

```diff
@@ -95,10 +96,18 @@ impl RelationalDB {
             let mut last_commit_offset = None;
             let mut last_hash: Option<Hash> = None;
             if let Some(message_log) = &message_log {
-                let message_log = message_log.lock().unwrap();
+                let mut message_log = message_log.lock().unwrap();
                 let max_offset = message_log.open_segment_max_offset;
-                for commit in commit_log::Iter::from(message_log.segments()) {
-                    let commit = commit?;
+                'commits: for commit in commit_log::Iter::from(message_log.segments()) {
+                    let commit = match commit {
+                        Ok(commit) => commit,
+                        Err(e) if e.kind() == io::ErrorKind::InvalidData => {
+                            log::warn!("Corrupt commit after offset {last_commit_offset:?}");
+                            message_log.reset_to(last_commit_offset.unwrap_or(0))?;
```

Comment:
Since this is destructive, I'm wondering if we want to add more safeguards against data loss here. And if so, which ones.

We could, for example, only reset the log if the offset falls into the last segment, and/or is within a certain distance from `open_segment_max_offset` (for some definition of "certain distance").